### ISSUE TITLE
Proposition: Settle on a Text Template for the TF Manual

### ISSUE THREAD

**Issue by Mik-TF** [threefoldtech/info_grid#62]
# Proposition: Text Template for the Manual

Here is a proposition to elaborate collectively a text template for the manual.

We currently have a guide on how to contribute to the manual: https://manual.grid.tf/contribute/contribute.html

We could add a section: `Main Template for TF Manual Contribution`, where we would state the general aesthetics and architectures of TF Manual guides and posts.

Once the template is settled and agreed by all, the TF manual will have a cohesive look, and it will be easier to collaborate and add content. We will then be able to focus as much as possible on creating content.

The main goal of this issue is to discuss and find how Threefold tech wants to present information in the TF Manual.

# Inspiration for this issue

The inspiration for this proposition is rooted in this pull request: https://github.com/threefoldtech/info_grid/pull/57, where @0oM4R and I share different views on how to present the information. This exchange is very fruitful.

By exchanging thoughts on the presentation with Omar, I already changed some way of presenting the information, as can be seen here: https://github.com/threefoldtech/info_grid/pull/61.

I would present a template here of what was discussed and other ideas, and would like everyone in this repo to share their input.
**Once we have the main template** of the tutorial itself, it will create a cohesion in the overall look of the manual, and it will give us more time to work on guides!

# Main Template of a TF Manual Markdown Text File

* single h1 for the heading 
* h2 for top level sections
* Can use `<h1>` for the heading instead of `#`, and `<h2>` instead of ## for the Table of contents title to permit shortcut as "Update TOC"
* Pictures
   * to show the click action, have a screenshot of the cursor preferably
* If your text reaches heading 4, you might want to separate your file into two or more sections of the guide

Any other ideas? Please share your thoughts!

# Propositions

## Left Side Interactive TOC

As @scottyeager shared, we could have some kind of interactive Table of Contents like Docker does [here](https://docs.docker.com/desktop/dev-environments/).

![Screen Shot 2023-03-30 at 9 01 57 PM](https://user-images.githubusercontent.com/77026219/228996799-ddce30ee-4680-4f70-a8ea-4be766d56a89.png)

## Embedded Videos and Video Platforms TF Content Sharing

Also, I would propose to aim at having videos for each guide, publish them on the Threefold youtube channel (and elsewhere), have direct links in the videos description to the Threefold Manual, and to embed those videos in the TF Manual sections.

This way, Threefold reaches a large audience (the youtube community, and other video communities) and also we offer different "ways" to learn the content (for visual type, reader type of learners, etc.).

We don't need fancy videos. We can have such videos of course. But just the simple screen recording of deployments, etc. can be sufficient. It generates content, reaches more audiences, and complement the text guides.

# Feedback

I would be more than happy to hear you guys on this. We can definitely learn and optimize the TF Manual if we can have everyone's input as we can see blind spots collectively.

Note: This proposition is based on discussions with @0oM4R as well as with @scottyeager.

**Comment by Mik-TF** [comment]
EDIT:

Here are the guidelines that could be proposed. We could add this in the section [How to Contribute](https://www.manual.grid.tf/contribute/contribute.html). This would facilitate help from the community.

***

# Main Template of a ThreeFold Manual Markdown File (.md)

Here are some suggestions on how to organize a markdown file when you submit contents to the ThreeFold Manual. This is not necessary, but it will ease the whole process.

* Title: Heading 1, h1, `#` 
* Main sections: Heading 2, h2, `##`
* For md files that contains a Table of Contents, 
   * Use `<h1>` instead of `#` for the _Title_ , and `<h2>` instead of `##` for the _Table of Contents_.
      * This quickens editing when creating and updating the ToC ([see this for more details](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one#table-of-contents)) 
   * Other heading labels should use standard markdown headings (`##`, etc.)
* Pictures
   * Not necessary nor recommended
      * It becomes hard to maintain the same quality and visual presentation throughout the manual
      * It takes more time and resources to update the pictures instead of texts
   * If you use pictures:
      * To show the click action, have a screenshot of the cursor preferably
      * For uniformity in the presentation, only take a screenshot without modifying the images if possible
* If your text reaches heading level 4, you might want to separate your file into two or more files
   *  A long article can be spread in many subsections

**Comment by Mik-TF** [comment]
@rkhamis 
Since we agreed on the basic of this issue, I updated the section how to contribute: https://www.manual.grid.tf/contribute/contribute.html#markdown-file-template-optional

We now have the main template for markdown files.

I did not include information on pictures/videos. As you said very well, texts are easier to maintain so I focused on texts for the md template.

Closing the issue. Thanks!
